src/core/task/tools/handlers/BrowserToolHandler.ts

```diff
@@ -75,7 +72,6 @@ export class BrowserToolHandler implements IFullyManagedTool {
 			config.taskState.consecutiveMistakeCount++
 			const errorResult = await config.callbacks.sayAndCreateMissingParamError("browser_action", "action")
 			await config.services.browserSession.closeBrowser()
-			await config.callbacks.saveCheckpoint()
```

Comment:
saveCheckpoint is called whenever a tool is executed, so calling it within the execute function is redundant. This is what caused the double checkpoint issue @0xToshii was investigating previously